test/shopify_api/throttled_test.exs

```diff
@@ -14,6 +14,53 @@ defmodule Test.ShopifyAPI.ThrottledTest do
   def func, do: send(self(), :func_called)
   def sleep_impl(_), do: send(self(), :sleep_called)
 
+  defmodule TrackerMock do
+    def get(_, _), do: {100, 0}
+    def api_hit_limit(_, _), do: {100, 0}
+
+    def update_api_call_limit(_, _) do
+      send(self(), :update_api_call_limit_called)
+      {100, 0}
+    end
+  end
+
+  describe "request/6" do
+    test "recurses when func returns over limit status code" do
+      func = fn ->
+        send(self(), :func_called)
+        {:ok, %{status_code: RateLimiting.REST.over_limit_status_code()}}
+      end
+
+      max_tries = 2
+      Throttled.request(func, @token, max_tries, TrackerMock)
+      for _ <- 1..max_tries, do: assert_received(:func_called)
+    end
+
+    test "recurses when func returns graphql throttled response" do
+      func = fn ->
+        send(self(), :func_called)
+        {:error, %{body: %{"errors" => [%{"message" => "Throttled"}]}, status_code: 200}}
+      end
+
+      max_tries = 2
+      Throttled.request(func, @token, max_tries, TrackerMock)
+      for _ <- 1..max_tries, do: assert_received(:func_called)
+    end
+
+    test "updates api call limit and does not recurse when func returns success" do
+      func = fn ->
+        send(self(), :func_called)
+        {:ok, %{status_code: 200}}
+      end
+
+      Throttled.request(func, @token, TrackerMock)
+
+      assert_receive :update_api_call_limit_called
+      assert_receive :func_called
+      refute_receive :func_called
+    end
+  end
+
```

Comment:
Added these tests to ensure I wasn't breaking the existing REST functionality :D